mulog-kinesis/src/com/brunobonacci/mulog/publishers/kinesis.clj

```diff
@@ -0,0 +1,71 @@
+(ns com.brunobonacci.mulog.publishers.kinesis
+  (:require [com.brunobonacci.mulog.buffer :as rb]
+            [com.brunobonacci.mulog.utils :as ut]
+            [com.brunobonacci.mulog.publishers.aws-utils :as awsutils]
+            [cheshire.core :as json]
+            [cheshire.generate :as gen]))
+
+;;
+;; Add Flake encoder to JSON generator
+;;
+(gen/add-encoder com.brunobonacci.mulog.core.Flake
+                 (fn [x ^com.fasterxml.jackson.core.JsonGenerator json]
+                   (gen/write-string json ^String (str x))))
+
+(defn- put-records
+  [{:keys [kinesis-client-params stream-name partition-key-name format] :as config} records]
+  (let [kinesis-client (awsutils/create-kinesis-client kinesis-client-params)
+        key-field partition-key-name
+        fmt* (if (= :json format) json/generate-string ut/edn-str)]
+    (->> records
+         (map (juxt #(str (get % key-field)) fmt*))
+         (map (fn [[k v]] (awsutils/create-records k v)))
+         (run! (partial awsutils/publish! kinesis-client stream-name)))))
```

Comment:
if you use `run!` then each record will be sent individually while it is better to send them in a batch and use the `:PutRecords` API fully.

Reply:
agree